ctriface/iface.go

```diff
@@ -536,7 +552,25 @@ func (o *Orchestrator) LoadSnapshot(ctx context.Context, vmID string, snap *snap
 		}
 	}()
 
+	logger.Debug("TEST: CreatVM request sent")
 	if o.GetUPFEnabled() {
+
+		logger.Debug("TEST: Registering VM with the memory manager")
+
+		stateCfg := manager.SnapshotStateCfg{
+			VMID:             vmID,
+			GuestMemPath:     o.getMemoryFile(vmID),
+			BaseDir:          o.getVMBaseDir(vmID),
+			GuestMemSize:     int(conf.MachineCfg.MemSizeMib) * 1024 * 1024,
+			IsLazyMode:       o.isLazyMode,
+			VMMStatePath:     o.getSnapshotFile(vmID),
+			WorkingSetPath:   o.getWorkingSetFile(vmID),
+			InstanceSockAddr: o.uffdSockAddr,
+		}
+		if err := o.memoryManager.RegisterVM(stateCfg); err != nil {
```

Comment:
hmm I am surprised that registration was not there. I am quite sure it is/was done inside the memory manager